device/cuda/src/ambiguity_resolution/kernels/scan_block_offsets.cu

```diff
@@ -0,0 +1,61 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "../../utils/barrier.hpp"
+#include "../../utils/global_index.hpp"
+#include "scan_block_offsets.cuh"
+
+// VecMem include(s).
+#include <vecmem/containers/device_vector.hpp>
+
+namespace traccc::cuda::kernels {
+
+__global__ void scan_block_offsets(device::scan_block_offsets_payload payload) {
+
+    if (*(payload.terminate) == 1 || *(payload.n_updated_tracks) == 0) {
+        return;
```

Comment:
I noticed this only now. Many of these kernels are implemented with undefined behaviour at the moment. :frowning: (One should not early return from kernels that require synchronization between the threads.)